db/migrate/20210709042135_create_external_upload_stubs.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class CreateExternalUploadStubs < ActiveRecord::Migration[6.1]
+  def change
+    create_table :external_upload_stubs do |t|
+      t.string :key, null: false
```

Comment:
Does the key column store the URL of the external upload? Does the key need to be unique if it is so?

Reply:
Well not the full URL, just the path to the upload on S3. But yeah probably makes sense for it to be unique; I do use a securerandom.hex call when creating the key as well, but I will add the unique index.